### PR TITLE
Week 13 assignment URL Shortner

### PR DESCRIPTION
**Pull request by joeshajan**
## Description
- Created a URL shorter app with react

## Type of Change
  
- [x] assignments completed
- [x] included readme file

### Where should the reviewer start? :checkered_flag:

on branch week13

`cd Week-13`

`npm install`

`npm start`

`npm run test`

### working screen recording of the app
https://www.loom.com/share/0e49fc6829934ba6b7ce8bbcb6d1f4dc

### CODE REVIEW

File: Week-13/url-shortner/src/utils/custom-hooks/useFetch.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+
+  const fetchData = async (url: string) => {
+    if (!url) return;
+    setIsLoading(true);
+    try {
+      const response = await fetch(url);
+
+      if (response.status === 400) return setError('Invalid Url');
```
Comment (rahulm2) [inline Week-13/url-shortner/src/utils/custom-hooks/useFetch.tsx:14]: why have u only checked for 400?

**Comment by rahulm2** [comment]
@joeshajan please attach a video of the working as well

File: Week-13/url-shortner/src/utils/custom-hooks/useFetch.tsx (hunk 1 above)
Comment (joeshajan) [reply Week-13/url-shortner/src/utils/custom-hooks/useFetch.tsx:14]: I added a new condition to check if the status is ok.
Is that correct?

File: Week-13/url-shortner/src/components/short-url/ShortUrl.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  const [data, isLoading, error, fetchData] = useFetch();
+
+  const onChangeHandler = (e: React.FormEvent<HTMLInputElement>) => {
+    setUrlInput(e.currentTarget.value);
+  };
+
+  const onSubmitHandler = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    fetchData(`https://api.shrtco.de/v2/shorten?url=${urlInput}`);
```
[the remark this answers is not in the document]
Comment (joeshajan) [reply Week-13/url-shortner/src/components/short-url/ShortUrl.tsx]: done

File: Week-13/url-shortner/src/components/short-url/ShortUrl.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import React, { memo, useEffect, useState } from 'react';
+
+import { useFetch } from '../../utils';
+import UrlContainer from '../UrlContainer';
+import Button from '../button/Button';
+
+type Props = {
```
[the remark this answers is not in the document]
Comment (joeshajan) [reply Week-13/url-shortner/src/components/short-url/ShortUrl.tsx]: done

should I keep one single types file for all the components or seperate types file for each component?

File: Week-13/url-shortner/src/components/short-url/ShortUrl.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+const ShortUrl = ({ addNewShortendUrl }: Props) => {
+  const [urlInput, setUrlInput] = useState('');
+  const [data, isLoading, error, fetchData] = useFetch();
+
+  const onChangeHandler = (e: React.FormEvent<HTMLInputElement>) => {
+    setUrlInput(e.currentTarget.value);
+  };
+
+  const onSubmitHandler = async (e: React.FormEvent<HTMLFormElement>) => {
```
[the remark this answers is not in the document]
Comment (joeshajan) [reply Week-13/url-shortner/src/components/short-url/ShortUrl.tsx]: done

File: Week-13/url-shortner/src/pages/Home.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { useCallback, useState } from 'react';
+
+import { Header, ShortUrl, ShortendUrl } from '../components';
+
+type Props = {};
+
+const Home = (props: Props) => {
+  const [urls, SetUrls] = useState<
```
[the remark this answers is not in the document]
Comment (joeshajan) [reply Week-13/url-shortner/src/pages/Home.tsx]: done

File: Week-13/url-shortner/src/utils/custom-hooks/useFetch.tsx (hunk 1 above)
Comment (rahulm2) [reply Week-13/url-shortner/src/utils/custom-hooks/useFetch.tsx:14]: yup

File: Week-13/url-shortner/src/components/short-url/ShortUrl.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (rahulm2) [reply Week-13/url-shortner/src/components/short-url/ShortUrl.tsx]: separate ones

**Pull request merged**